cirq-superstaq/cirq_superstaq/service.py

```diff
@@ -98,6 +98,54 @@ def counts_to_results(
     return result
 
 
+def _validate_cirq_circuit(circuits: Any) -> None:
+    """Validates that the input is either a single `cirq.Circuit` or a list of `cirq.Circuit`
+    instances.
+
+    Args:
+        circuits: The circuit to run.
+
+    Raises:
+        ValueError: If the input is not a `cirq.Circuit` or a list of
+        `cirq.Circuit` instances.
+    """
+
+    if not (
+        isinstance(circuits, cirq.Circuit)
+        or (
+            isinstance(circuits, Sequence)
+            and all(isinstance(circuit, cirq.Circuit) for circuit in circuits)
+        )
+    ):
+        raise ValueError(
+            "Invalid 'circuits' input. Must be a `cirq.Circuit` or a "
+            "sequence of `cirq.Circuit` instances."
+        )
+
+
+def _validate_get_counts(repetitions: Any) -> None:
```

Comment:
can we give this a more generic name
```suggestion
def _validate_num_repetitions(repetitions: object) -> None:
```
and use it to validate `num_eca_circuits` as well? (if so we should also change the error messages to something more generic, e.g. `TypeError(f"{repetitions} cannot be safely cast as an integer.")`)